src/ratatui/hello-rust/mousefood.md

```diff
@@ -0,0 +1,318 @@
+# Hello, Rat
+
+In this chapter, we'll write a simple program to get started with Ratatui. We won't do anything fancy here, just the basics to see how it works.
+
+## Prerequisites
+
+- You'll need a TFT Display for this chapter. If you haven't completed the [TFT Display chapter](./tft-display/index.md), I recommend finishing that first and then coming back here. Since the circuit connections between the TFT Display and ESP32 are already explained there, we won't repeat those instructions here.​
+
+- Ratatui has a nice [tutorial to get started](https://ratatui.rs/tutorials/hello-ratatui/) for building TUI apps. If you already know Ratatui, this exercise is pretty simple - just gluing things together. If you're new, check out the official Ratatui tutorials later to get familiar with the basics and build better UIs.
+
+
+## Generate project using esp-generate
+
+To create the project, use the `esp-generate` command. Run the following:
+
+```sh
+esp-generate --chip esp32 hello-rat
+```
+
+This will open a screen asking you to select options. Ratatui requires heap allocation, so we will enable "unstable" and "alloc" features. 
+
+- First, select the option "Enable unstable HAL features."
+- Select the option "Enable allocations via the esp-alloc crate."
+
+Optionally, you can enable the logging feature also
+
+- So, scroll to "Flashing, logging and debugging (espflash)" and hit Enter.
+- Then, Select "Use defmt to print messages".
+
+Just save it by pressing "s" in the keyboard.
+
+## Dependencies
+
+First, let's update the Cargo.toml file with the dependencies we need to control the TFT display. We've already gone over these earlier in the TFT display section. 
+
+```toml
+embedded-hal-bus = { version = "0.3" }
+display-interface-spi = "0.5"
+ili9341 = "0.6.0"
```

Comment:
Someone on Discord asked would this tutorial work if they had a different driver. We should probably note this. e.g. `ili9341` driver is used here but you can also use another display driver according to your hardware etc.